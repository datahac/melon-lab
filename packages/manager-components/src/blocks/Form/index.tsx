import React, { StatelessComponent } from 'react';

export interface FormProps {
  onSubmit?: () => any;
}

const Form: StatelessComponent<FormProps> = ({ children, onSubmit }) => (
  <form onSubmit={onSubmit} className="form">
    {children}
  </form>
);

export default Form;
