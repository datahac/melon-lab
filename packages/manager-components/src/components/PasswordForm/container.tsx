import { withFormik } from 'formik';
import { compose, withHandlers } from 'recompose';
import * as Yup from 'yup';

import PasswordForm from './index';

const withFormValidation = withFormik({
  mapPropsToValues: props => ({ ...props.initialValues }),
  validationSchema: Yup.object().shape({
    password: Yup.string().required('Password is required.'),
  }),
  enableReinitialize: true,
  handleSubmit: (values, form) => {
    if (form.props.onSubmit) {
      form.props.onSubmit(values, 'passwordForm');
    }
  },
});

const withFormHandler = compose(
  withHandlers({
    onChange: props => (values, event) => {
      props.setFieldValue(event.target.name, values.value);
    },
  }),
);

export default compose(
  withFormValidation,
  withFormHandler,
)(PasswordForm);
