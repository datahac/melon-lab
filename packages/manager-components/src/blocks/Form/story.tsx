import { action } from '@storybook/addon-actions';
import { storiesOf } from '@storybook/react';
import React from 'react';
import Button from '../Button/index';
import Form from './index';

storiesOf('Blocks|Form', module).add('Default', () => {
  return (
    <Form onSubmit={action('submit')}>
      <Button type="submit">Submit</Button>
    </Form>
  );
});
