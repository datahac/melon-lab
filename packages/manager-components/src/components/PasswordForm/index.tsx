import React, { StatelessComponent } from 'react';
import Button from '~/blocks/Button';
import Form from '~/blocks/Form';
import Input from '~/blocks/Input';

import styles from './styles.css';

interface FormValues {
  password: string;
}

export interface PasswordFormProps {
  values: FormValues;
  handleSubmit: () => void;
  handleReset: () => void;
  onCancel: () => void;
  handleBlur?: () => void;
  onChange?: () => void;
  touched?: any;
  errors?: any;
}

export const PasswordForm: StatelessComponent<PasswordFormProps> = ({
  values,
  handleBlur,
  onChange,
  touched,
  errors,
  handleSubmit,
  onCancel,
  handleReset,
}) => {
  const handleCancel = () => {
    handleReset();
    onCancel();
  };

  return (
    <div className="password-form">
      <style jsx>{styles}</style>
      <Form>
        <div className="password-form__input">
          <Input
            value={values.password}
            type="password"
            placeholder="Password"
            name="password"
            onChange={onChange}
            onBlur={handleBlur}
            required={true}
            error={touched.password && errors.password}
          />
        </div>
        <div className="password-form__buttons">
          <div className="password-form__button">
            <Button type="button" style="secondary" onClick={handleCancel}>
              Cancel
            </Button>
          </div>
          <div className="password-form__button">
            <Button type="submit" onClick={handleSubmit}>
              Confirm
            </Button>
          </div>
        </div>
      </Form>
    </div>
  );
};

export default PasswordForm;
