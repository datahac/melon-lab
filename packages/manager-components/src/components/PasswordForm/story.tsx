import { storiesOf } from '@storybook/react';
import React from 'react';
import PasswordForm from './container';

const initialProps = {
  initialValues: {
    password: '',
  },
};

storiesOf('Components|Password Form', module).add('Default', () => {
  return <PasswordForm {...initialProps} />;
});
