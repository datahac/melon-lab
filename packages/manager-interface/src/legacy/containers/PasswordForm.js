import { connect } from 'react-redux';
import PasswordForm from '@melonproject/manager-components/components/PasswordForm/container';
import { actions } from '../actions/modal';

const mapStateToProps = state => ({
  initialValues: {
    password: '',
  },
});

const mapDispatchToProps = dispatch => ({
  onSubmit: values => {
    dispatch(actions.passwordEntered(values.password));
  },
  onCancel: () => {
    dispatch(actions.cancel());
  }
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(PasswordForm);
