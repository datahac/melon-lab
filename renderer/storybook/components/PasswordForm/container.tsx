import { withFormik } from 'formik';
import * as Yup from 'yup';

import PasswordForm from './index';

const withFormValidation = withFormik({
  mapPropsToValues: props => ({ ...props.initialValues }),
  validationSchema: Yup.object().shape({
    password: Yup.string()
      .required('Password is required.')
      .min(8, 'Password needs to be at least 8 chars long. For your security!'),
  }),
  enableReinitialize: true,
  handleSubmit: (values, form) =>
    form.props.onSubmit && form.props.onSubmit(values, 'passwordForm'),
});

export default withFormValidation(PasswordForm);
