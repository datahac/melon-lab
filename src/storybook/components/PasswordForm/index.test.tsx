import React from 'react';
import Form from './';
import PasswordForm from './container';

const onSubmit = jest.fn();
const onCancel = jest.fn();

const data = {
  initialValues: {
    password: '',
  },
  onCancel,
  onSubmit,
};

describe('PasswordForm', () => {
  const defaultElement = <PasswordForm {...data} />;
  let tree;

  beforeEach(() => {
    tree = mount(defaultElement);
  });

  it('should render correctly', () => {
    expect(tree).toMatchSnapshot();
  });

  it('should submit the form if valid', async () => {
    tree.setProps({ initialValues: { password: 'password' } });
    await tree
      .find(Form)
      .props()
      .submitForm();
    expect(onSubmit).toHaveBeenCalledWith({ password: 'password' }, 'passwordForm');
  });

  it('should call onCancel event', () => {
    tree
      .find(Form)
      .find('Button')
      .first()
      .simulate('click');
    expect(onCancel).toHaveBeenCalled();
  });
});
