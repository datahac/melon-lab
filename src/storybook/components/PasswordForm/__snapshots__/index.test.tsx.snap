// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordForm should render correctly 1`] = `
<WithFormik(Function)
  initialValues={
    Object {
      "password": "",
    }
  }
  onCancel={[MockFunction]}
  onSubmit={[MockFunction]}
>
  <Formik
    enableReinitialize={true}
    handleSubmit={[Function]}
    initialValues={
      Object {
        "password": "",
      }
    }
    isInitialValid={false}
    onCancel={[MockFunction]}
    onSubmit={[Function]}
    render={[Function]}
    validateOnBlur={true}
    validateOnChange={true}
    validationSchema={[Function]}
  >
    <Component
      dirty={false}
      errors={Object {}}
      handleBlur={[Function]}
      handleChange={[Function]}
      handleReset={[Function]}
      handleSubmit={[Function]}
      initialValues={
        Object {
          "password": "",
        }
      }
      isSubmitting={false}
      isValid={false}
      isValidating={false}
      onCancel={[MockFunction]}
      onSubmit={[MockFunction]}
      registerField={[Function]}
      resetForm={[Function]}
      setError={[Function]}
      setErrors={[Function]}
      setFieldError={[Function]}
      setFieldTouched={[Function]}
      setFieldValue={[Function]}
      setFormikState={[Function]}
      setStatus={[Function]}
      setSubmitting={[Function]}
      setTouched={[Function]}
      setValues={[Function]}
      submitCount={0}
      submitForm={[Function]}
      touched={Object {}}
      unregisterField={[Function]}
      validateField={[Function]}
      validateForm={[Function]}
      validateOnBlur={true}
      validateOnChange={true}
      values={
        Object {
          "password": "",
        }
      }
    >
      <div
        className="jsx-__scopedHash password-form"
      >
        <JSXStyle
          css="__scoped"
          styleId="__scopedHash"
        />
        <Form
          onSubmit={[Function]}
        >
          <form
            className="form"
            onSubmit={[Function]}
          >
            <div
              className="jsx-__scopedHash password-form__input"
            >
              <Input
                name="password"
                onBlur={[Function]}
                onChange={[Function]}
                placeholder="Password"
                required={true}
                type="password"
                value=""
              >
                <div
                  className="jsx-__scopedHash input"
                >
                  <JSXStyle
                    css="__scoped"
                    styleId="__scopedHash"
                  />
                  <input
                    className="jsx-__scopedHash input__field"
                    name="password"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                  <div
                    className="jsx-__scopedHash input__error"
                  />
                </div>
              </Input>
            </div>
            <div
              className="jsx-__scopedHash password-form__buttons"
            >
              <div
                className="jsx-__scopedHash password-form__button"
              >
                <Button
                  onClick={[Function]}
                  style="secondary"
                  type="button"
                >
                  <button
                    className="jsx-__scopedHash button button--medium button--secondary"
                    onClick={[Function]}
                    type="button"
                  >
                    <JSXStyle
                      css="__scoped"
                      styleId="__scopedHash"
                    />
                    Cancel
                  </button>
                </Button>
              </div>
              <div
                className="jsx-__scopedHash password-form__button"
              >
                <Button
                  type="submit"
                >
                  <button
                    className="jsx-__scopedHash button button--medium button--primary"
                    onClick={[Function]}
                    type="submit"
                  >
                    <JSXStyle
                      css="__scoped"
                      styleId="__scopedHash"
                    />
                    Confirm
                  </button>
                </Button>
              </div>
            </div>
          </form>
        </Form>
      </div>
    </Component>
  </Formik>
</WithFormik(Function)>
`;
